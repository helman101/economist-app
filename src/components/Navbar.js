import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { logUserOutAction } from '../actions/userActions';

const Navbar = (props) => {
  const handleLogOut = () => {
    props.dispatch(logUserOutAction());
  };

  return (
    <nav className="navbar bg-white w-100 border">
      <div className="logo d-flex bg-danger text-white flex-column align-items-center py-2 px-2 fw-bold ms-3">
        <div>
          The
        </div>
        <div>
          Economist
        </div>
      </div>
      <div className="me-3">
        <Link to="/" className="btn btn-danger" type="button" onClick={handleLogOut}>LOG OUT</Link>
      </div>
    </nav>
  );
};

Navbar.propTypes = {
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
  dispatch: PropTypes.func.isRequired,
};

export default connect()(Navbar);
