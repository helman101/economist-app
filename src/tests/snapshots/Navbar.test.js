import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import defaultState from '../../store/defaultStore';
import Navbar from '../../components/Navbar';

const mockStore = configureStore([]);

describe('Navbar', () => {
  let store;
  let component;
  let history;

  beforeEach(() => {
    history = createMemoryHistory();
    store = mockStore(defaultState);
    component = renderer.create(
      <Provider store={store}>
        <Navbar history={history} />
      </Provider>,
    );
  });

  it('should render propertly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
