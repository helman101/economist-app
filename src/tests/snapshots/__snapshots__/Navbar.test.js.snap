// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render propertly 1`] = `
<nav
  className="navbar bg-white w-100 border"
>
  <div
    className="logo d-flex bg-danger text-white flex-column align-items-center py-2 px-2 fw-bold ms-3"
  >
    <div>
      The
    </div>
    <div>
      Economist
    </div>
  </div>
  <div
    className="me-3"
  >
    <a
      className="btn btn-danger"
      href="/"
      onClick={[Function]}
      type="button"
    >
      LOG OUT
    </a>
  </div>
</nav>
`;
